enarx-keep-sev-shim/src/gdt.rs

```diff
@@ -0,0 +1,174 @@
+            tss.privilege_stack_table[0] = {
+                static mut STACK: [Page; STACK_NUM_PAGES] = [Page::zeroed(); STACK_NUM_PAGES];
+                stack_end(VirtAddr::from_ptr(&STACK), STACK_NUM_PAGES)
+            };
+
+            // The stacks for the exceptions and interrupts
+            tss.interrupt_stack_table[0] = {
+                static mut STACK: [Page; STACK_NUM_PAGES] = [Page::zeroed(); STACK_NUM_PAGES];
+                stack_end(VirtAddr::from_ptr(&STACK), STACK_NUM_PAGES)
+            };
+            tss.interrupt_stack_table[1] = {
+                static mut STACK: [Page; STACK_NUM_PAGES] = [Page::zeroed(); STACK_NUM_PAGES];
+                stack_end(VirtAddr::from_ptr(&STACK), STACK_NUM_PAGES)
+            };
+            tss.interrupt_stack_table[2] = {
+                static mut STACK: [Page; STACK_NUM_PAGES] = [Page::zeroed(); STACK_NUM_PAGES];
+                stack_end(VirtAddr::from_ptr(&STACK), STACK_NUM_PAGES)
+            };
+            tss.interrupt_stack_table[3] = {
+                static mut STACK: [Page; STACK_NUM_PAGES] = [Page::zeroed(); STACK_NUM_PAGES];
+                stack_end(VirtAddr::from_ptr(&STACK), STACK_NUM_PAGES)
+            };
+            tss.interrupt_stack_table[4] = {
+                static mut STACK: [Page; STACK_NUM_PAGES] = [Page::zeroed(); STACK_NUM_PAGES];
+                stack_end(VirtAddr::from_ptr(&STACK), STACK_NUM_PAGES)
+            };
+            tss.interrupt_stack_table[5] = {
+                static mut STACK: [Page; STACK_NUM_PAGES] = [Page::zeroed(); STACK_NUM_PAGES];
+                stack_end(VirtAddr::from_ptr(&STACK), STACK_NUM_PAGES)
+            };
+            tss.interrupt_stack_table[6] = {
+                static mut STACK: [Page; STACK_NUM_PAGES] = [Page::zeroed(); STACK_NUM_PAGES];
+                stack_end(VirtAddr::from_ptr(&STACK), STACK_NUM_PAGES)
+            };
+            tss
+        });
+    }
+
+    unsafe {
+        GDT = Some({
+            let mut gdt = GlobalDescriptorTable::new();
+
+            // `syscall` loads segments from STAR MSR assuming a data_segment follows `kernel_code_segment`
+            // so the ordering is crucial here. Star::write() will panic otherwise later.
+            let code_selector = gdt.add_entry(Descriptor::kernel_code_segment());
+            let data_selector = gdt.add_entry(Descriptor::UserSegment(
+                (DescriptorFlags::USER_SEGMENT
+                    | DescriptorFlags::PRESENT
+                    | DescriptorFlags::WRITABLE
+                    | DescriptorFlags::LONG_MODE)
+                    .bits(),
+            ));
+
+            // `sysret` loads segments from STAR MSR assuming `user_code_segment` follows `user_data_segment`
+            // so the ordering is crucial here. Star::write() will panic otherwise later.
+            let mut user_data_selector = gdt.add_entry(Descriptor::user_data_segment());
+            user_data_selector.set_rpl(PrivilegeLevel::Ring3);
```

Comment:
I think this flag is already set ([x86_64/gdt.rs:221](https://github.com/rust-osdev/x86_64/blob/master/src/structures/gdt.rs#L221))

Reply:
true... was fixed lately